website/cue/reference/components/sinks/axiom.cue

```diff
@@ -0,0 +1,134 @@
+package metadata
+
+components: sinks: axiom: {
+	title: "Axiom"
+
+	classes: {
+		commonly_used: false
+		delivery:      "at_least_once"
+		development:   "beta"
+		egress_method: "batch"
+		service_providers: ["Axiom"]
+		stateful: false
+	}
+
+	features: {
+		acknowledgements: true
+		healthcheck: enabled: true
+		send: {
+			batch: {
+				enabled:      true
+				common:       false
+				max_events:   1000
+				max_bytes:    1_048_576
+				timeout_secs: 1.0
+			}
+			compression: enabled: false
+			encoding: {
+				enabled: true
+				codec: enabled: false
+			}
+			proxy: enabled: true
+			request: {
+				enabled: true
+				headers: true
+			}
+			tls: {
+				enabled:                true
+				can_verify_certificate: true
+				can_verify_hostname:    true
+				enabled_default:        false
+			}
+			to: {
+				service: services.axiom
+
+				interface: {
+					socket: {
+						api: {
+							title: "Axiom API"
+							url:   urls.axiom
+						}
+						direction: "outgoing"
+						protocols: ["http"]
+						ssl: "required"
+					}
+				}
+			}
+		}
+	}
+
+	support: {
+		requirements: []
+		warnings: []
+		notices: []
+	}
+
+	configuration: {
+		token: {
+			description: "Your Axiom token"
+			required:    true
+			warnings: []
+			type: string: {
+				examples: ["xxxx", "${AXIOM_TOKEN}"]
+				syntax: "literal"
+			}
+		}
+		dataset: {
+			description: "Your Axiom dataset"
+			required:    true
+			warnings: []
+			type: string: {
+				examples: ["vector.dev"]
+				syntax: "literal"
+			}
+		}
+		url: {
+			description: "Your Axiom URL (only required if not Axiom Cloud)"
+			common:      false
+			required:    false
+			warnings:    []
+			type: string: {
+				examples: ["https://cloud.axiom.co", "${AXIOM_URL}"]
+				syntax: "literal"
+			  default:     ""
+			}
+		}
+		org_id: {
+			description: "Your Axiom Org ID (only required for personal tokens)"
+			common:      false
+			required:    false
+			warnings:    []
+			type: string: {
+				examples: ["xxxx", "${AXIOM_ORG_ID}"]
+				syntax: "literal"
+			  default:     ""
+			}
+		}
+	}
+
+	input: {
+		logs: true
+		metrics: null
+		traces: true
```

Comment:
Does the same endpoint support sending traces?

Reply:
Yes, they're in the same format right? Axiom is really flexible and people use us for all kinds of things including logs and traces.

Reply:
Gotcha, yeah, our traces follow the OpenTelemetry data model so if that works for Axiom, then this seems fine. You will need to update the sink's `fn input()` to return `Input::log() | Input::trace()` though. Currently Vector will fail to boot if a trace source is hooked up to this sink.

Reply:
Along the same line, the `elasticsearch` sink supports metrics by just transforming them into a log event before sending them - so we could do that here as well

Reply:
I think we're fine with supporting logs and metrics for now with the `axiom` sink. I'm happy to implement `trace_to_log` in a separate PR though 🙌